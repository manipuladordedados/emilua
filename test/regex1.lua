local byte_span = require 'byte_span'
local regex = require 'regex'

local re = regex.new{
    pattern = 'a',
    grammar = 'extended'
}

print(regex.match(re, ''))
print(regex.match(re, 'a'))
print(regex.match(re, 'a '))
print(regex.match(re, ' a'))

print(regex.match(re, byte_span.new(0)))
print(regex.match(re, byte_span.append('a')))
print(regex.match(re, byte_span.append('a ')))
print(regex.match(re, byte_span.append(' a')))

re = regex.new{
    pattern = 'a.*',
    grammar = 'extended'
}

print(regex.match(re, ''))
print(regex.match(re, 'a'))
print(regex.match(re, 'a '))
print(regex.match(re, ' a'))
