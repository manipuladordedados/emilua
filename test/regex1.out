nil
a
nil
nil
nil
a
nil
nil
nil
a
a 
nil
